rtloader/test/datadog_agent/datadog_agent.go

```diff
@@ -356,3 +358,16 @@ func obfuscateMongoDBString(cmd *C.char, errResult **C.char) *C.char {
 		return nil
 	}
 }
+
+//export emitAgentTelemetry
+func emitAgentTelemetry(check *C.char, metric *C.char, value C.float, metricType *C.char) {
+	checkName := C.GoString(check)
+	metricName := C.GoString(metric)
+	metricValue := float64(value)
```

Comment:
Since golang doesn't have a native `double` we're going to have to cast into float at some point anyway aren't we? In practice I doubt that level of precision will ever matter for these metrics.

Reply:
> Since golang doesn't have a native double

It does, you are using it here. C `float` corresponds to `float32` in Go.